frontend/src/components/Forms/MineRecordForm.js

```diff
@@ -43,6 +76,21 @@ export class MineRecordForm extends Component {
       mine_commodity_code: [],
       mine_disturbance_code: [],
     };
+    // when a currentMineType is deleted, update this.state.usedTenureTypes
+    if (this.props.currentMineTypes && this.props.currentMineTypes !== nextProps.currentMineTypes) {
+      const prevTenure = this.props.currentMineTypes.map(
+        ({ mine_tenure_type_code }) => mine_tenure_type_code
+      );
+      const nextTenure = nextProps.currentMineTypes.map(
+        ({ mine_tenure_type_code }) => mine_tenure_type_code
+      );
+      const diff = _.difference(prevTenure, nextTenure);
+      this.setState((prevState) => ({
+        usedTenureTypes: prevState.usedTenureTypes.filter(
+          (tenureTypes) => !diff.includes(tenureTypes)
+        ),
+      }));
+    }
     // Do nothing if no mine_types, or no change to mine_types
     if (!nextProps.mine_types || nextProps.mine_types === this.props.mine_types) {
       return;
```

Comment:
This looks like an early exit. It should likely be above your other logic

Reply:
removed the entire block.